File: Patterns/Adapter Pattern/Cpp/Turkey/TurkeyAdapter.cpp
#include "./Turkey.cpp"
#include "../Duck/Duck.cpp"

class TurkeyAdapter : public Duck {
private:
    Turkey* turkey;

public:
    TurkeyAdapter(Turkey* turkey) : turkey{turkey} {}

    void quack() override {
        turkey->gobble();
    }

    void fly() override {
        for (int i = 0; i < 5; i++) {
            turkey->fly();
        }
    }
};

File: Patterns/Adapter Pattern/CSharp/Turkey/TurkeyAdapter.cs
using Duck;

namespace Turkey
{
    public class TurkeyAdapter : Duck.Duck
    {
        private Turkey turkey;

        public TurkeyAdapter(Turkey turkey)
        {
            this.turkey = turkey;
        }

        public void Quack() => turkey.Gobble();

        public void Fly()
        {
            for (var i = 0; i < 5; i++)
            {
                turkey.Fly();
            }
        }
    }
}